ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLChatAgentRunner.java

```diff
@@ -167,8 +167,13 @@ private void runAgent(MLAgent mlAgent, Map<String, String> params, ActionListene
                     executeParams.put(key.replace(toolSpec.getType() + ".", ""), params.get(key));
                 }
             }
-            log.info("Fetching tool for type: " + toolSpec.getType());
-            Tool tool = toolFactories.get(toolSpec.getType()).create(executeParams);
+            String toolType = toolSpec.getType();
+            log.info("Fetching tool for type: " + toolType);
+
+            if (toolFactories.get(toolType) == null) {
+                log.error(toolType + " can't be found, please use ListTool API to check if it's deployed.");
```

Comment:
Same thing here, Please use pattern instead of manual concatenation of text in log message:
`log.error("{} can't be found, please use ListTool API to check if it's deployed.", toolType);`